src/utils/intersection-observer-polyfill.js

```diff
@@ -100,8 +101,15 @@ export function getIntersectionChangeEntry(element, owner, hostViewport) {
  * @return {boolean}
  */
 export function nativeIntersectionObserverSupported(win) {
+  if (!('IntersectionObserver' in win)) {
+    return false;
+  }
+  // eslint-disable-next-line no-undef
+  if (INTERSECTION_OBSERVER_POLYFILL || getMode().localDev || getMode().test) {
```

Comment:
nativeInOb is considered supported even when the connection between inabox and its host failed. 
I am a bit worried that it will change the behavior in that case. 
https://github.com/ampproject/amphtml/blob/97099ce8bf4826a08d964e7a02da55a7641b1090/extensions/amp-analytics/0.1/visibility-manager.js#L761-L767
the old polyfill tick with `this.vewport_getRect()` once, what's the behavior with INTERSECTION_OBSERVER_POLYFILL inabox host connection failure?

Reply:
The default behavior: all observers are blocked. It's very easy to unblock them, e.g. when connection fails. It should just go the polyfill extension. We just need to figure out the timeout. Do we really want 0 timeout for connection issues?

Reply:
Checked the code again. LGTM, the reason we want tick with 0 timeout is also to polyfill the `observe` method, to invoke the callback when it is registered. 
It should change the behavior here because the initial `inaboxViewport.boxRect_` is placed below the window, so that the visibility event won't be triggered.

This is the ideal solution based on the comment here
https://github.com/ampproject/amphtml/blob/d94fa740ed6f9ab098248b207f5c161654752ebe/src/inabox/inabox-viewport.js#L457-L460

Reply:
The InOb polyfill is definitely more desirable. But could you clarify? Should we timeout host connection here? I.e. would it be better to report wrong values or not report them at all? And if it's a timeout thing, what should it be?

Reply:
give me one sec to verify the behavior when `visiblePercentageMin=0`

Reply:
IMO we don't need to timeout unless we want to do error reporting. 

I do think it's better to not report compared to report wrong values. That's the case today already with the only caveat that an event `[visiblePercentageMin: 0, visiblePercentageMax: 0]` (element is not visible) will still be triggered when the connection fails. And this PR will fix it.

Reply:
Ok. Sounds like this is resolved then.